chapter13/dtj/ex7

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Getopt::Std;
+use File::Basename;
+
+our($opt_s);
+getopts('s');
+my $link_function = $opt_s ? \&CORE::symlink : \&CORE::link;
+die "Need 2 arguments, source and destination" unless @ARGV == 2;
```

Comment:
I did test this and it *seems* to, not quite sure what would happen if you gave it a switch that it didn't support. Then again, sometimes we don't always know what modifies `@ARGV` so perhaps I should have been more careful.